### PR TITLE
Minimal replacement of active/crystallized state with single state

### PR DESCRIPTION
**Pull request by vbuterin**
Depends on https://github.com/ethereum/eth2.0-specs/pull/120.

Does not include deeper simplifications that become possible with this approach; this is a fairly simple search and replace.

### CODE REVIEW

**Comment by prestonvanloon** [comment]
Context? Why make this change?

**Comment by vbuterin** [comment]
It's a protocol simplification (two states down to one) that makes way for other protocol simplifications (adjusting RANDAO instantly instead of at the end of the cycle, among other things), which was not viable before because of re-hashing costs, but is viable now, because of the SSZ tree hashing proposal: https://github.com/ethereum/eth2.0-specs/pull/120

File: specs/beacon-chain.md [hunk 1]
```diff
@@ -668,22 +649,22 @@ For each one of these attestations:
...
 
 Additionally, verify and update the RANDAO reveal. This is done as follows:
 
 * Let `repeat_hash(x, n) = x if n == 0 else repeat_hash(hash(x), n-1)`.
-* Let `V = crystallized_state.validators[curblock_proposer_index]`.
-* Verify that `repeat_hash(block.randao_reveal, (block.slot - V.randao_last_reveal) // RANDAO_SLOTS_PER_LAYER + 1) == V.randao_commitment`, and set `active_state.randao_mix = xor(active_state.randao_mix, block.randao_reveal)` and append to `ActiveState.pending_specials` a `SpecialObject(kind=RANDAO_CHANGE, data=[bytes8(curblock_proposer_index), block.randao_reveal])`.
+* Let `V = state.validators[curblock_proposer_index]`.
+* Verify that `repeat_hash(block.randao_reveal, (block.slot - V.randao_last_reveal) // RANDAO_SLOTS_PER_LAYER + 1) == V.randao_commitment`, and set `state.randao_mix = xor(state.randao_mix, block.randao_reveal)` and append to `State.pending_specials` a `SpecialObject(kind=RANDAO_CHANGE, data=[bytes8(curblock_proposer_index), block.randao_reveal])`.
```
Comment (djrtwo) [inline specs/beacon-chain.md]: Should we consider removing the `RANDAO_CHANGE` special record and just allow the one validator record to be updated?

This kind of optimization can occur in a separate PR

Comment (djrtwo) [reply specs/beacon-chain.md]: Ah, I just saw your comment above. Deeper changes to follow.

File: specs/beacon-chain.md [hunk 2]
```diff
@@ -170,24 +168,7 @@ A `SpecialRecord` has the following fields:
...
-    'pending_specials': [SpecialRecord]
-    # recent beacon block hashes needed to process attestations, older to newer
-    'recent_block_hashes': ['hash32'],
-    # RANDAO state
-    'randao_mix': 'hash32'
-}
-```
-
-The `CrystallizedState` has the following fields:
+The `state` has the following fields:
```
[the remark this answers is not in the document]
Comment (vbuterin) [reply specs/beacon-chain.md]: Done.

**Pull request merged**